aten/src/THC/generic/THCTensor.cpp

```diff
@@ -100,20 +100,21 @@ THCTensor *THCTensor_(newWithStorage1d)(THCState *state, THCStorage *storage, pt
     c10::intrusive_ptr<at::StorageImpl>::reclaim(new_storage),
     at::DispatchKey::CUDATensorId
   ).release();
-  THTensor_(setStorageNd)(self, storage, storageOffset, 1,
-                          &size0, &stride0);
+  THTensor_(setStorageNd)(state, self, storage, storageOffset, 1, &size0, &stride0);
 
   return self;
 }
 
-THCTensor *THCTensor_(newWithSize)(THCState *state, at::IntArrayRef size, at::IntArrayRef stride)
-{
-  return THCTensor_(newWithStorage)(state, NULL, 0, size, stride);
-}
-
 THCTensor *THCTensor_(newWithSize1d)(THCState *state, int64_t size0)
 {
-  return THCTensor_(newWithSize)(state, {size0}, {});
+  THCStorage *new_storage = THCStorage_(new)(state);
+  THCTensor *self = c10::make_intrusive<at::TensorImpl, at::UndefinedTensorImpl>(
+    c10::intrusive_ptr<at::StorageImpl>::reclaim(new_storage),
+    at::DispatchKey::CUDATensorId
+  ).release();
+  THCTensor_(setStorageNd)(state, self, 0, 1, &size0, nullptr);
```

Comment:
Another pattern-match observation, `THTensor` version passes `new_storage` rather than `self` as second arg

Reply:
good catch, although the "second arg" thing is because THCTensor always have extra state.  But I was indeed neglecting to pass new_storage.